### PR TITLE
Updates to build with the latest SDK

### PR DESCRIPTION
**Pull request by juanjux**
This fix some of the fields detected by the new check and explicitly ignores others, with comments and link to https://github.com/bblfsh/python-driver/issues/178 in those cases.

### CODE REVIEW

File: driver/normalizer/normalizer.go [hunk 1]
```diff
@@ -28,31 +28,53 @@ var Normalize = Transformers([][]Transformer{
...
+			{Name: "decorator_list", Op: Var("func_decorators")},
 		},
 		Obj{
 			"Nodes": Arr(
 				Obj{
 					// FIXME: generator=true if it uses yield anywhere in the body
 					"async": Bool(async),
 				},
+				Obj{
+					"decorators": Var("func_decorators"),
```
Comment (dennwc) [inline driver/normalizer/normalizer.go]: Can be added to the object above.

File: driver/normalizer/normalizer.go [hunk 2]
```diff
@@ -28,31 +28,53 @@ var Normalize = Transformers([][]Transformer{
...
 					"Name": UASTType(uast.Identifier{}, Obj{
 						"Name": Var("name"),
 					}),
 					"Node": UASTType(uast.Function{}, Obj{
-						"Type": UASTType(uast.FunctionType{}, Obj{
-							"Arguments": Var("arguments"),
+						"Type": UASTType(uast.FunctionType{}, Fields{
+							{Name: "Arguments", Op: Var("arguments")},
+							{Name: "Returns", Optional: "ret_opt", Op: Cases("ret_case",
+								Is(nil),
```
Comment (dennwc) [inline driver/normalizer/normalizer.go]: Python functions always have an implicit return, so should be a single empty `Argument` with `Init` = `None` when there are no type annotations. There is a similar piece of code in [JS](https://github.com/bblfsh/javascript-driver/blob/master/driver/normalizer/normalizer.go#L274)).

File: driver/normalizer/normalizer.go [hunk 3]
```diff
@@ -88,8 +110,8 @@ func mapStr(nativeType string) Mapping {
 			{Name: uast.KeyType, Op: String("Boxed" + nativeType)},
 			{Name: "boxed_value", Op: UASTType(uast.String{}, Obj{
 				uast.KeyPos: Var("pos_"),
-				"Value": Var("s"),
-				"Format": String(""),
+				"Value":     Var("s"),
+				"Format":    String(""),
```
Comment (dennwc) [inline driver/normalizer/normalizer.go]: Not related to the change, but zero values may be omitted.

File: driver/normalizer/normalizer.go [hunk 4]
```diff
@@ -165,6 +208,11 @@ var Normalizers = []Mapping{
 	AnnotateType("keyword", MapObj(
 		Fields{
 			{Name: "arg", Op: Var("name")},
+			// TODO: change this once we've a way to store other nodes on semantic objects
```
Comment (dennwc) [inline driver/normalizer/normalizer.go]:
```suggestion
			// FIXME: change this once we've a way to store other nodes on semantic objects
```

File: driver/normalizer/normalizer.go [hunk 5]
```diff
@@ -174,35 +222,40 @@ var Normalizers = []Mapping{
...
 	MapSemantic("arg", uast.Argument{}, MapObj(
-		Obj{
-			uast.KeyToken: Var("name"),
+		Fields{
+			{Name: uast.KeyToken, Op: Var("name")},
+			{Name: "default", Optional: "opt_def", Op: Var("init")},
+			// No problem dropping this one, it's used by an internal interpreter optimization/cache
+			// without semantic meaning
+			{Name: "ctx", Optional: "opt_ctx", Op: Any()},
+			// TODO: change this once we've a way to store other nodes on semantic objects
```
Comment (dennwc) [inline driver/normalizer/normalizer.go]:
```suggestion
			// FIXME: change this once we've a way to store other nodes on semantic objects
```
a bit more comments like this below

File: driver/normalizer/normalizer.go (hunk 3 above)
Comment (juanjux) [reply driver/normalizer/normalizer.go]: Done in next commit.

File: driver/normalizer/normalizer.go (hunk 5 above)
Comment (juanjux) [reply driver/normalizer/normalizer.go]: Done (all) in next commit.

File: driver/normalizer/normalizer.go (hunk 1 above)
Comment (juanjux) [reply driver/normalizer/normalizer.go]: Done in next commit (plus function comments in the same Obj).

File: driver/normalizer/normalizer.go (hunk 2 above)
Comment (juanjux) [reply driver/normalizer/normalizer.go]: I implemented this and it worked beautifully... but then I noticed that this annotations only has a value IF the function has a type annotation BUT if it doesn't have it, it doesn't mean that the function returns None! Just that it's not annotated. So I think is better to left it with the current state (nil if there is no annotation, the annotation if there is one).

**Comment by juanjux** [comment]
Ready for another review round!

File: driver/normalizer/normalizer.go (hunk 2 above)
Comment (dennwc) [reply driver/normalizer/normalizer.go]: I haven't meant that the function returns `None` as type, only that if there is no return, all Python2 and Python3 functions returns `None` implicitly. So it's `Returns.Argument.Init: None`, same way as we do it in JS with `undefined`.

File: driver/normalizer/normalizer.go [hunk 6]
```diff
@@ -143,6 +172,29 @@ var Normalizers = []Mapping{
 		}),
 	),
 
+	Map(
+		Part("_", Fields{
+			{Name: uast.KeyType, Op: String("Attribute")},
+			{Name: uast.KeyPos, Op: Var("pos_")},
+			{Name: "attr", Op: Var("aname")},
+			// No problem dropping this one, it's used by an internal interpreter optimization
+			//cache
```
Comment (bzz) [inline driver/normalizer/normalizer.go]:
```suggestion
			// cache without semantic meaning
```

Formatting nit pick

File: driver/normalizer/normalizer.go (hunk 2 above)
Comment (juanjux) [reply driver/normalizer/normalizer.go]: Yes, but to know if the function returns None implicitly we should parse the function and find the return statement in this case.

**Pull request merged**